### DISCUSSION TITLE
Downloading of source code

### DISCUSSION THREAD

**Discussion by rdori** [mui/toolpad#1546] [Help and support] [answered]
Does MUI toolpad allow you to download the React / JS source code that would generated for each component or page you generate via the GUI? I would like to use the toolpad to help with quick/easy design, styling, and layout of components (including some custom components), but then include these library components into a larger React app that I have.

**Comment by oliviertassinari** [comment, marked as answer]
Hi, this is something that we consider in https://github.com/mui/mui-toolpad/issues/316. However, I think that it's unlikely that we will go down this route, if ever. Why? the problem we are trying to solve isn't to help developers write more code. The best line of code is the one that never needs to be written. This is still very true https://www.youtube.com/watch?v=GnO7D5UaDig&t=2451s, 26 years later.

Instead, would it make sense for you to have Toolpad React components? It would be like any other React components, it could import any other components. This type of component would be YAML, and editable with Toolpad visual environment running locally?
